### ISSUE TITLE
`Organize Imports` and `Fix All` do nothing in helix

### ISSUE THREAD

**Issue by bluthej** [astral-sh/ruff-lsp#249]
Hey there, I've been using helix for a few weeks now and I'm using `ruff-lsp` alongside `pylsp` to hack on some Python code :)
Thanks so much to the maintainers!

I do like to keep my imports tidy, but for some reason it seems like the `Organize Imports` code action does nothing in helix, and `Fix All` as well. Removing unused imports works just fine of the other hand. I know it sounds like #28 but since it is closed I'm thinking whatever wasn't right at the time got fixed.

I was wondering if that is a helix issue? I went over to the helix repo but I couldn't find any mention of such a limitation.

Here is some additional info:

- `ruff-lsp` version: 0.0.39
- `helix` version: built from the master branch on the 20th of September 

<details><summary>helix lsp config for Python</summary>
<p>

```toml
[language-server.ruff-lsp]
command = "ruff-lsp"
args = []

[[language]]
name = "python"
formatter = { command = "black", args = ["--quiet", "-"] }
auto-format = true
language-servers = ["ruff-lsp", "pylsp"]
```

</p>
</details> 

<details><summary>helix log</summary>
<p>

```bash
2023-09-20T22:31:11.028 helix_term::commands::lsp [DEBUG] code action: CodeAction { title: "Ruff: Organize Imports", kind: Some(CodeActionKind("source.organizeImports")), diagnostics: Some([]), edit: None, command: None, is_preferred: None, disabled: None, data: Some(String("file:///home/joffrey/test.py")) }
```

</p>
</details>

**Comment by bluthej** [comment]
Ok so after playing around a bit more I found out that if I select the I001 rule in my ruff settings then I have a separate `Ruff (I001): Organize imports` code action that **does** sort my imports, in addition to the `Ruff: Organize Imports` code action that's always there and which does nothing.

I know this sounds a bit like #91, but my point here is that I find it odd that the always-present `Ruff: Organize Imports` code action does nothing, even though the rule is selected. Also, the `Ruff: Fix All` code action does nothing, even though there are things to fix. Could that be an issue on the helix side or is everything acting properly and I just have misplaced expectations?

As a user I just find it a bit confusing, and if I can, I would like to spare future users some frustration :smile: 
So if there is anything I can do to help out I'd be happy to :smiley:

**Comment by charliermarsh** [comment]
Sorry for not getting back to you earlier, and thanks for the kind words.

There are probably two things happening here:

1. We always show the `Ruff: Organize Imports` and `Ruff: Fix All` actions, even if there's nothing to organize or nothing to fix. This can lead to a confusing experience. It makes sense for VS Code, but perhaps not for other LSPs, and we may want to filter them out contextually.
2. Separately, though, it sounds like those actions aren't doing what you'd expect in Helix, which I need to test.

**Comment by charliermarsh** [comment]
Hmm, so after testing, things are actually working correctly for me -- the code actions have the expected effect. And surprisingly, we're only showing `Ruff: Organize Imports` if there are imports to fix (same for `Ruff: Fix All`) -- I'd forgotten that we _did_ make it contextual for editors that don't support deferred resolution of actions, so I think (1) shouldn't apply to Helix at least.

<img width="948" alt="Screen Shot 2023-09-26 at 7 34 43 PM" src="https://github.com/astral-sh/ruff-lsp/assets/1309177/47e43829-b4a2-4835-b870-c27aa3d533fd">

<img width="948" alt="Screen Shot 2023-09-26 at 7 34 47 PM" src="https://github.com/astral-sh/ruff-lsp/assets/1309177/b13e802a-9fed-45a3-9281-7c57e76dfbbb">

<img width="948" alt="Screen Shot 2023-09-26 at 7 34 50 PM" src="https://github.com/astral-sh/ruff-lsp/assets/1309177/27bf41b1-4aeb-4f18-bbd7-7f9652476413">

<img width="948" alt="Screen Shot 2023-09-26 at 7 35 01 PM" src="https://github.com/astral-sh/ruff-lsp/assets/1309177/8c626d32-7f9c-4588-bb4f-7600391d438d">

**Comment by charliermarsh** [comment]
I don't know that it will resolve it, but you could try using a single language server?

```toml
[[language]]
name = "python"
scope = "source.python"
language-server = { command = "ruff-lsp" }
config = { settings = { args = [] } }
```

**Comment by bluthej** [comment]
Fixed by [helix-editor/helix#8421](https://github.com/helix-editor/helix/pull/8421).
